app/services/external_upload_manager.rb

```diff
@@ -59,17 +84,16 @@ def promote_to_upload!
     UploadCreator.new(tempfile, external_upload_stub.original_filename, opts).create_for(
       external_upload_stub.created_by_id
     )
+  rescue SizeMismatchError
+    external_upload_stub.destroy!
```

Comment:
Is there a reason we don't just destroy this before raising the error? Having to rescue here seems kind of odd.

Reply:
I did this because we are marking the stub as failed for other errors. They need to be failed because the `abort_multipart` endpoint still needs them to be around, HOWEVER, thinking about this it would be much better to just delete the upload stub unconditionally here when an error occurs and abort the multipart upload at the same time (if the stub is multipart), that way the client doesn't have to make another trip to abort-multipart for this upload.

Will do some refactoring to make this so. Will be cleaner this way, the Manager deletes + aborts files where an error occurs on the last step.

Reply:
Ah disregard this; the call to promote_to_upload! happens when the multipart upload is already complete, so we just have to delete the file in S3, not abort. Still going to change this to just delete the stub though.